### PR TITLE
test(e2e): Add test that connects to target using go ssh library

### PR DESCRIPTION
**Pull request by moduli**
This PR adds an e2e test that uses the [go ssh library](https://pkg.go.dev/golang.org/x/crypto/ssh) to connect to a `tcp` target.

There will be a follow-up PR in `boundary-enterprise` to add a similar test for `ssh` targets.

### CODE REVIEW

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	require.NoError(t, err)
+	signer, err := ssh.ParsePrivateKey(privateKeyRaw)
+	require.NoError(t, err)
+
+	config := &ssh.ClientConfig{
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		User:            c.TargetSshUser,
+		Auth:            []ssh.AuthMethod{ssh.PublicKeys(signer)},
+	}
+	client, err := ssh.Dial("tcp", fmt.Sprintf("localhost:%s", port), config)
```
Comment (moduli) [inline testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:69]: Not sure if `tcp` is the right option here

**Review by tmessi — APPROVED** [review]

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	boundary.WaitForSessionCli(t, ctx, newProjectId)
+
+	// Connect to the target using the golang ssh library
+	privateKeyRaw, err := os.ReadFile(c.TargetSshKeyPath)
+	require.NoError(t, err)
+	signer, err := ssh.ParsePrivateKey(privateKeyRaw)
+	require.NoError(t, err)
+
+	config := &ssh.ClientConfig{
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```
[the remark this answers is not in the document]
Comment (tmessi) [reply testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:65]: This makes sense here, especially for a test.

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go (hunk 1 above)
Comment (tmessi) [reply testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:69]: That looks correct.

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/hashicorp/boundary/internal/target"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+)
+
+// TestCliTcpTargetConnectGolangSsh uses the golang ssh library to connect to a target
+func TestCliTcpTargetConnectGolangSsh(t *testing.T) {
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:22]: Pettiest of complaints but I dislike the use of `golang` generally as a phrase. These set of repos are generally known as "experimental" or "sub-repositories": https://pkg.go.dev/golang.org/x. The `golang` in the URL is out of necessity, the name is Go. What do you think about
```suggestion
// TestCliTcpTargetConnectGoSsh uses the go experimental ssh library to connect to a target
func TestCliTcpTargetConnectGoSsh(t *testing.T) {
```

Would obviously mean renaming the file too.

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/hashicorp/boundary/internal/target"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:18]: Is this intentionally using 3 import groups? I think we generally use 2.

**Review by johanbrandhorst — APPROVED** [review]

File: testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go (hunk 4 above)
Comment (moduli) [reply testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go:18]: Weird. Don't know how that happened there.

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/base/target_tcp_connect_golang_ssh_test.go]: Some import formatters do this

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**